app/src/main/java/com/duckduckgo/app/browser/userAgent/UserAgentProvider.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2018 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.userAgent
+
+import android.net.Uri
+import android.os.Build
+import javax.inject.Inject
+
+
+fun Uri.isMobileSite() : Boolean = this.authority.startsWith("m.")
+
+fun Uri.toDesktopUri(): Uri {
+    return if(this.isMobileSite()) {
+        Uri.parse(toString().replaceFirst("m.", ""))
+    } else {
+        this
+    }
+}
+
+/**
+ * Example Default User Agent (From Chrome):
+ * Mozilla/5.0 (Linux; Android 8.1.0; Nexus 6P Build/OPM3.171019.014) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.137 Mobile Safari/537.36
+ *
+ * Example Default Desktop User Agent (From Chrome):
+ * Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.137 Safari/537.36
+ */
+class UserAgentProvider @Inject constructor() {
+
+    companion object {
```

Comment:
Can we place companion objects at the bottom? I got the same feedback in my PR :) https://kotlinlang.org/docs/reference/coding-conventions.html#source-code-organization